libnetwork/sandbox_externalkey_unix.go

```diff
@@ -131,7 +131,7 @@ func (c *Controller) acceptClientConnections(sock string, l net.Listener) {
 		conn, err := l.Accept()
 		if err != nil {
 			if _, err1 := os.Stat(sock); os.IsNotExist(err1) {
-				log.G(context.TODO()).Debugf("Unix socket %s doesn't exist. cannot accept client connections", sock)
+				log.G(context.TODO()).Debugf("Unix socket %s was closed. The external key listener will stop.", sock)
```

Comment:
That one has been bothering me for a long time, because this appears on shutdown I think?

Been wondering if we could actually _fix_ that; if I recall correctly, during shutdown we call some code that's also executed on _startup_. Wondering if we can make a better flow so that it doesn't even _attempt_ to do this.

(this was part of cleaning up at shutdown?)

Reply:
> That one has been bothering me for a long time, because this appears on shutdown I think?
> 
> Been wondering if we could actually fix that; if I recall correctly, during shutdown we call some code that's also executed on startup. Wondering if we can make a better flow so that it doesn't even attempt to do this.
> 
> (this was part of cleaning up at shutdown?)

Yeah, this only happens during daemon shutdown:

* `(*Daemon).Shutdown()` calls libnet's `(*Controller).Stop()` ;
* Which then calls `(*Controller).stopExternalKeyListener()` ;
* Which then call the `Close()` method of the `net.Listener` `extKeyListener` ;
* In turn, this method unlink and close the UNIX socket ;